collect_app/src/main/java/org/odk/collect/android/logic/AuditEvent.java

```diff
@@ -40,24 +40,43 @@
         FINALIZE_ERROR("finalize error"),                                           // Error in finalize
         CONSTRAINT_ERROR("constraint error"),                                       // Constraint or missing answer error on save
         DELETE_REPEAT("delete repeat"),                                             // Delete a repeat group
-        GOOGLE_PLAY_SERVICES_NOT_AVAILABLE("google play services not available"),   // Google Play Services are not available
-        LOCATION_PERMISSIONS_GRANTED("location permissions granted"),               // Location permissions are granted
-        LOCATION_PERMISSIONS_NOT_GRANTED("location permissions not granted"),       // Location permissions are not granted
-        LOCATION_TRACKING_ENABLED("location tracking enabled"),                     // Location tracking option is enabled
-        LOCATION_TRACKING_DISABLED("location tracking disabled"),                   // Location tracking option is disabled
-        LOCATION_PROVIDERS_ENABLED("location providers enabled"),                   // Location providers are enabled
-        LOCATION_PROVIDERS_DISABLED("location providers disabled"),                 // Location providers are disabled
-        UNKNOWN_EVENT_TYPE("Unknown AuditEvent Type");                              // Unknown event type
+
+        // Google Play Services are not available
+        GOOGLE_PLAY_SERVICES_NOT_AVAILABLE("google play services not available", true),
+        // Location permissions are granted
+        LOCATION_PERMISSIONS_GRANTED("location permissions granted", true),
+        // Location permissions are not granted
+        LOCATION_PERMISSIONS_NOT_GRANTED("location permissions not granted", true),
+        // Location tracking option is enabled
+        LOCATION_TRACKING_ENABLED("location tracking enabled", true),
+        // Location tracking option is disabled
+        LOCATION_TRACKING_DISABLED("location tracking disabled", true),
+        // Location providers are enabled
+        LOCATION_PROVIDERS_ENABLED("location providers enabled", true),
+        // Location providers are disabled
+        LOCATION_PROVIDERS_DISABLED("location providers disabled", true),
+        // Unknown event type
+        UNKNOWN_EVENT_TYPE("Unknown AuditEvent Type");
 
         private final String value;
+        private final boolean isLocationRelated;
 
-        AuditEventType(String value) {
+        AuditEventType(String value, boolean isLocationRelated) {
```

Comment:
I think we should fallow the same approach we use to determine if it's an internal event. We don't add a new parameter to the constructor but we have a special method: https://github.com/opendatakit/collect/pull/3175/files#diff-b25f62214ddaa8161b77f62ab938dcaeR122

It would be better to have similar `isLocationAuditEventType` method

Reply:
I agree it would be good to group everything to do with audit event types together. Ideally those would be properties on the event type entity. `isIntervalAuditEventType` is written as an instance method but really it's static (doesn't have anything to do with the instance) which generally indicates an incomplete design. My preference would be to add an `isInterval` boolean property to the `AuditEventType` enum. Why would you prefer to have static methods at the `AuditEvent` level?

Reply:
Ok sounds fine, I just want both cases to be consistent so we can fallow your approach.

Reply:
88c7949